### ISSUE TITLE
Data Augmentation of uint16 images (such as PNG)

### ISSUE THREAD

**Issue by aziabari** [tensorflow/tensorflow#12528]
How we can do data augmentation (rotation, flip,...) of uint16 images?

Here is a piece of my code:

import os
import tensorflow as tf

filQ=['ex4497_fbp_13.png','ex4497_fbp_12.png']
filQQ=tf.train.string_input_producer(filQ)

reader =tf.WholeFileReader()
key, value = reader.read(filQQ)

myimg = tf.image.decode_png(value, dtype=tf.uint16) 
myimg1 = tf.image.rot90(myimg,k=1,name=None)

I get an error that:

TypeError: Value passed to parameter 'tensor' has DataType uint16 not in list of allowed values: uint8, int8, int32, int64, bool, float16, float32, float64, complex64, complex128, string

So basically unit16 doesn't exist!
unit8 is not a good choice because the pixel values in my image are NOT in the range [0 255],
and we can only use data type uint8, uint16 for PNG in tensorflow!!

So again the question becomes:
**How we can do data augmentation (rotation, flip,...) of uint16 images?**

**Comment by yongtang** [comment]
The issue comes from the uint16 support of `rot90` (`reverse`). The uint16 and int16 are already available in CPU. Added a PR #12531 to enable the uint16 and int16 support of `reverse`.

**Comment by aziabari** [comment]
Thank you! I am not sure what to do next now that you added this PR. 
Would you please let me know what should I do?

**Comment by yongtang** [comment]
@aziabari once the PR #12531 is merged your issue will be resolved.

**Comment by aziabari** [comment]
I am new to Python/TF. Could you please let me know what do you mean by merging PR #12531?
Do I need to update anything or make any changes on my computer?

**Comment by yongtang** [comment]
@aziabari The necessary code change that is needed to fixed the issue you encountered has been made. In the next release of TensorFlow (I don't know but I would guess 1.4) you can update your local TensorFlow installment and the issue you encountered on you local computer would be gone.

I am not sure what is the next release date of TensorFlow (1.4) though it may take some time.

In the mean time, you can also try building TensorFlow from source: https://www.tensorflow.org/install/install_sources

That will allows you to update your local TensorFlow installment without waiting for the official release of 1.4.

**Comment by aziabari** [comment]
Thanks!
